app/routes.py

```diff
@@ -1,2 +1,225 @@
-from flask import Blueprint
+from flask import Blueprint, request, make_response, jsonify
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import datetime
+from sqlalchemy import DateTime, desc
+import requests
+import os
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+#wave4
+def hi_slack_api(task):
+    key = os.environ.get("API_KEY")
+    url = "https://slack.com/api/chat.postMessage"
+    slack_str = f"Someone just completed the task {task.title}"
+    requests.post(url, headers={"Authorization": key}, params={"channel": "task-list-api", "text": slack_str})
+
+#wave6
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes= False)
+def task_goal(goal_id):
+    request_body = request.get_json()
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return make_response("", 404)
+
+    for task_id in request_body["task_ids"]:
+        task = Task.query.get(task_id)
+        task.goal_id = goal.goal_id
+ 
+    db.session.commit()   
+    return make_response({"id": goal.goal_id, "task_ids": request_body["task_ids"]}, 200)
```

Comment:
This change would allow the response to include ALL tasks associated with the goal, not just the new ones.
This isn't super intuitive so I'd definitely be down to talk through this more in our next 1:1


```suggestion
     task_list = []
    for task in goal.tasks:
        task_list.append(task.task_id)
        
    return make_response({"id": goal.goal_id, "task_ids": task_list}, 200)                                                                          
```